torch/distributed/_composable/contract.py

```diff
@@ -33,6 +32,7 @@ class RegistryItem:
 
 
 _TState = TypeVar("_TState", bound="_State", covariant=True)
+M = TypeVar("M", nn.Module, list[nn.Module])
```

Comment:
Did you intend for the typevar to be public?

Reply:
Hmm that's a good question. I see where you are coming from and I can make this private to be consistent with existing ones. That said, I think in general the question is should the typing info here all be public as they are inherently exposed as part of the public APIs.